asf_welsh_energy_consultation/getters/get_data.py

```diff
@@ -485,3 +503,73 @@ def load_wales_hp(wales_epc):
     wales_hp = wales_epc.loc[wales_epc.HP_INSTALLED].reset_index(drop=True)
 
     return wales_hp
+
+
+def load_mcs_epc_combined():
+    """
+    Get combined gold MCS-EPC dataset filtered for rows with heat pump installations in domestic dwellings. Use local preprocessed dataset unless specified
+    to download data from S3. Downloaded data goes through pre-processing to produce desired pd.DataFrame.
+
+    Returns:
+        pd.DataFrame: Gold MCS-EPC dataset for domestic dwellings with heat pumps.
+    """
+    args = get_args()
+    batch = args.gold_mcs_epc_batch
+    download_data = args.download_gold_data_from_s3
+
+    if not download_data:
+        path = os.path.join(input_data_path, f"hp_installed_gold_{batch}.csv")
+        return pd.read_csv(path)
+
+    else:
+        path = f"outputs/gold/merged_epc_mcs_installations_installers_{batch}.csv"
+
+        logger.info(f"Loading {path} from S3. This will take a while.")
+
+        download_from_s3(path_to_file=path, output_path=input_data_path)
+
+        ddf = dd.read_csv(
+            os.path.join(
+                input_data_path,
+                f"merged_epc_mcs_installations_installers_{batch}.csv",
+            ),
+            dtype={
+                "HP_INSTALL_DATE": "object",
+                "UPRN": "object",
+                "installation_type": "object",
+            },
+        )
+
+        # Get rows with HP installed only, data already filtered for domestic only
+        hp_installed = ddf[ddf["HP_INSTALLED"] == True]
+        hp_installed = hp_installed[
+            [
+                "POSTCODE",
+                "INSPECTION_DATE",
+                "COUNTRY",
+                "UPRN",
+                "HP_INSTALLED",
+                "HP_TYPE",
+                "HP_INSTALL_DATE",
+                "MCS_AVAILABLE",
+                "EPC_AVAILABLE",
+            ]
+        ]
+
+        hp_installed = hp_installed.rename(columns={"POSTCODE": "postcode"})
+
+        # Convert to pandas df
+        df = hp_installed.compute()
+
+        df["HP_INSTALL_DATE"] = pd.to_datetime(df["HP_INSTALL_DATE"])
+
+        # Batch 231009 contains data from MCS up to 30 June 2023 and data from EPC up to 31 July 2023
+        # Must remove additional month of EPC data for consistency
+        if batch == "231009":
+            df = df[df["HP_INSTALL_DATE"] < "2023-07-01"]
```

Comment:
Instead you could just check max date in EPC, max date in MCS and get the min between the two max's :) it would avoid the hardcoding. Also because this might happen in future batches.

```suggestion
        max_epc_date = df["HP_INSTALL_DATE"].max()
        max_mcs_date = df["commision_date"].max()
        df = df[df["HP_INSTALL_DATE"] <= min(max_epc_date, max_mcs_date)]
```

Reply:
Great idea but it doesn't look like 'commission_date' col exists in the dataset. I believe the processing combines all installation/commission dates into the HP_INSTALL_DATE col